sql/src/main/java/org/apache/druid/sql/calcite/planner/convertlet/TimeInIntervalConvertletFactory.java

```diff
@@ -56,7 +56,7 @@ public class TimeInIntervalConvertletFactory implements DruidConvertletFactory
       .operandTypes(SqlTypeFamily.TIMESTAMP, SqlTypeFamily.CHARACTER)
       .requiredOperandCount(2)
       .literalOperands(1)
-      .returnTypeNonNull(SqlTypeName.BOOLEAN)
+      .returnTypeInference(ReturnTypes.BOOLEAN_NULLABLE)
```

Comment:
could you take a look at other methods not permitting `null`-on outputs?

it seems like that for example `regexp_like` is also affected:
```
        SELECT REGEXP_LIKE(dim1,'x') from foo where dim1 is null
```

Reply:
fixed in regexp_like, substring & contains operators. rest looked good to me.